### ISSUE TITLE
Support X-Gitlab-Token

### ISSUE THREAD

**Issue by lopossumi** [jenkinsci/generic-webhook-trigger-plugin#186]
Hi, and first of all thanks for the great plugin!

Unfortunately we're currently limited to using Generic Webhook Trigger for a single job in Jenkins, as GitLab sends their webhook tokens in their own way:
```
Request headers:
Content-Type: application/json
X-Gitlab-Event: Push Hook
X-Gitlab-Token: MyToken
```
Would it be possible to try parsing the token also from this source, or is there a workaround?

Related issue in GitLab has been open for two years, so I'm not expecting it to resolve any time soon: [Add ability to specify webhook authentication token header name](https://gitlab.com/gitlab-org/gitlab/-/issues/17290)

**Comment by tomasbjerre** [comment]
Released in 1.69. Open issue again if any problems!

**Comment by lopossumi** [comment]
Doesn't seem to work. I tested the post against a PostBin test server and it seems that the actual request header is written in lowercase (unlike what GitLab's own logs say), so that could be the reason:

**Headers**
x-real-ip: 111.222.333.444
host: postb.in
connection: close
content-length: 1586
content-type: application/json
x-gitlab-event: Push Hook
x-gitlab-token: MyToken

**Comment by tomasbjerre** [comment]
Fixed in 1.70.

**Comment by lopossumi** [comment]
Works like a charm! Thanks! ⭐
